src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.cs

```diff
@@ -6,72 +6,177 @@ namespace System.Numerics.Tensors
+        /// <para>
+        /// <paramref name="x"/> and <paramref name="y"/> may overlap, but neither may overlap with <paramref name="destination"/>; if they do, behavior is undefined.
+        /// </para>
+        /// <para>
+        /// If any of the element-wise input values is <see cref="float.NaN"/>, the resulting element-wise value is <see cref="float.NaN"/>.
+        /// </para>
+        /// <para>
+        /// This method may call into the underlying C runtime or employ instructions specific to the current architecture. Exact results may differ between different
+        /// operating systems or architectures.
+        /// </para>
+        /// </remarks>
         public static void AddMultiply(ReadOnlySpan<float> x, ReadOnlySpan<float> y, float multiplier, Span<float> destination) =>
             InvokeSpanSpanScalarIntoSpan<AddMultiplyOperator>(x, y, multiplier, destination);
 
-        /// <summary>Computes the element-wise result of: <c>(<paramref name="x" /> + <paramref name="y" />) * <paramref name="multiplier" /></c>.</summary>
+        /// <summary>Computes the element-wise result of <c>(<paramref name="x" /> + <paramref name="y" />) * <paramref name="multiplier" /></c> for the specified tensors.</summary>
         /// <param name="x">The first tensor, represented as a span.</param>
         /// <param name="y">The second tensor, represented as a scalar.</param>
         /// <param name="multiplier">The third tensor, represented as a span.</param>
         /// <param name="destination">The destination tensor, represented as a span.</param>
-        /// <exception cref="ArgumentException">Length of '<paramref name="x" />' must be same as length of '<paramref name="multiplier" />'.</exception>
+        /// <exception cref="ArgumentException">Length of <paramref name="x" /> must be same as length of <paramref name="multiplier" />.</exception>
         /// <exception cref="ArgumentException">Destination is too short.</exception>
-        /// <remarks>This method effectively does <c><paramref name="destination" />[i] = (<paramref name="x" />[i] + <paramref name="y" />) * <paramref name="multiplier" />[i]</c>.</remarks>
+        /// <remarks>
+        /// <para>
+        /// This method effectively computes <c><paramref name="destination" />[i] = (<paramref name="x" />[i] + <paramref name="y" />) * <paramref name="multiplier" />[i]</c>.
+        /// </para>
+        /// <para>
+        /// <paramref name="x"/> and <paramref name="multiplier"/> may overlap, but neither may overlap with <paramref name="destination"/>; if they do, behavior is undefined.
+        /// </para>
+        /// <para>
+        /// If any of the element-wise input values is <see cref="float.NaN"/>, the resulting element-wise value is <see cref="float.NaN"/>.
+        /// </para>
+        /// <para>
+        /// This method may call into the underlying C runtime or employ instructions specific to the current architecture. Exact results may differ between different
```

Comment:
AddMultiply and MultiplyAdd are cases we should determine if we want to guarantee determinism or if we want to give the flexibility to use a fused operation internally, this would give a more accurate result across both operations and be within 1 ULP of the two separate operations

I’d lean towards determinism and that we should expose a separate MultiplyAddEstimate method which allows the optimization via explicit opt-in, as we’ve done elsewhere for math apis

Reply:
Should I interpret this as "we should revert https://github.com/dotnet/runtime/pull/92205"?

Reply:
CosineSimilarity is one of the more complex APIs where we can’t easily guarantee it today, so I think it’s fine there.

That’s different from the more primitive API, like MultiplyAdd, where it’s trivial to make the determinism guarantee and where we don’t really lose any perf by doing so.

Reply:
Long term, I’d like us to move towards cross platform determinism everywhere, but that’s a long goal and not something to worry about for this release where differences are already guaranteed to exist.

So I think we only need to be concerned with the places where we can trivially provide perf and determinism